optaplanner-persistence/optaplanner-persistence-jackson/src/test/java/org/optaplanner/persistence/jackson/api/score/constraint/ConstraintMatchJacksonJsonSerializerAndDeserializerTest.java

```diff
@@ -0,0 +1,82 @@
+package org.optaplanner.persistence.jackson.api.score.constraint;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import org.junit.Before;
+import org.junit.Test;
+import org.optaplanner.core.api.score.Score;
+import org.optaplanner.core.api.score.buildin.hardsoft.HardSoftScore;
+import org.optaplanner.core.api.score.constraint.ConstraintMatch;
+import org.optaplanner.persistence.jackson.api.score.ScoreJacksonJsonSerializer;
+import org.optaplanner.persistence.jackson.api.score.buildin.hardsoft.HardSoftScoreJacksonJsonDeserializer;
+
+import static org.junit.Assert.assertEquals;
+
+public class ConstraintMatchJacksonJsonSerializerAndDeserializerTest {
+
+    ObjectMapper mapper;
+
+    @Before
+    public void setUp() throws Exception {
+        mapper = new ObjectMapper();
+        mapper.registerModule(new SimpleModule()
+                .addSerializer(HardSoftScore.class, new ScoreJacksonJsonSerializer<HardSoftScore>())
+                .addDeserializer(HardSoftScore.class, new HardSoftScoreJacksonJsonDeserializer()));
+    }
+
+    private void testSerializeAndDeserialize(String packageName, String name, Score score, Object... list) {
+        ConstraintMatch constraintMatch = new ConstraintMatch(packageName, name, Arrays.asList(list), score);
+        TestConstraintMatchWrapper wrapper = new TestConstraintMatchWrapper(constraintMatch);
+
+        try {
+            String json = mapper.writeValueAsString(wrapper);
+
+            TestConstraintMatchWrapper other = mapper.readValue(json, TestConstraintMatchWrapper.class);
+
+            assertEquals(constraintMatch, other.getConstraintMatch());
+        } catch (JsonParseException e) {
+            throw new IllegalStateException(e);
```

Comment:
interestingly enough, for test code, we are totally fine with test helper methods that have a long throws method signature so I wouldn't bother wrapping these.

Reply:
That somewhat contradicts lines 36 to 42 of `AbstractScoreJacksonJsonSerializerAndDeserializer`:
```
try {
            ObjectMapper objectMapper = new ObjectMapper();
            jsonString = objectMapper.writeValueAsString(input);
            output = (W) objectMapper.readValue(jsonString, input.getClass());
        } catch (IOException e) {
            throw new IllegalStateException("Marshalling or unmarshalling for input (" + input + ") failed.", e);
        }
```